pkg/cached/cache.go

```diff
@@ -226,40 +241,36 @@ func (c *listMerger[T, V]) Get() Result[V] {
 	return c.result
 }
 
-// NewTransformer creates a new cache that transforms the result of
-// another cache. The transformFn will only be called if the source
-// cache has updated the output, otherwise, the cached result will be
-// returned.
+// Transform the result of another cache. The transformFn will only be called
+// if the source cache has updated the output, otherwise, the cached result will
+// be returned.
 //
 // If the dependency returned an error before, or it returns an error
 // this time, or if the transformerFn failed before, the function is
 // reran.
-func NewTransformer[T, V any](transformerFn func(Result[T]) Result[V], source Data[T]) Data[V] {
-	return NewListMerger(func(caches []Result[T]) Result[V] {
+func Transform[T, V any](transformerFn func(Result[T]) Result[V], source Data[T]) Data[V] {
+	return MergeList(func(caches []Result[T]) Result[V] {
 		if len(caches) != 1 {
 			panic(fmt.Errorf("invalid cache for transformer cache: %v", caches))
 		}
 		return transformerFn(caches[0])
 	}, []Data[T]{source})
 }
 
-// NewSource creates a new cache that generates some data. This
-// will always be called since we don't know the origin of the data and
-// if it needs to be updated or not. sourceFn MUST be thread-safe.
-func NewSource[T any](sourceFn func() Result[T]) Data[T] {
-	c := source[T](sourceFn)
-	return &c
+// Func wraps a (thread-safe) function as a Data[T].
+func Func[T any](fn func() Result[T]) Data[T] {
+	return dataFunc[T](fn)
 }
 
-type source[T any] func() Result[T]
+type dataFunc[T any] func() Result[T]
 
-func (c *source[T]) Get() Result[T] {
-	return (*c)()
+func (c dataFunc[T]) Get() Result[T] {
+	return c()
 }
 
-// NewStaticSource creates a new cache that always generates the
+// Cache creates a new cache that always generates the
```

Comment:
Using this name `Cache` in a package called `cached` makes it a bit ambiguous. It's hard to tell what this function does without looking at the source code. This might also be misleading for the default way of using this package. I would probably prefer something like `StaticFunc` or `Once`.

Reply:
LazilyOnce ?

Reply:
Once is also fine. I like that.